example-adapter/src/main/java/org/niis/xrd4j/exampleadapter/ExampleAdapter.java

```diff
@@ -225,10 +290,152 @@ protected String deserializeRequest(Node requestNode, SOAPMessage message) throw
                     logger.debug("Found \"name\" element.");
                     // "name" element was found - return the text content
                     return requestNode.getChildNodes().item(i).getTextContent();
-                }
+                        }
             }
             logger.warn("No \"name\" element found. Null is returned.");
             return null;
         }
     }
+
+    /**
+     * This class is responsible for serializing the response data of the listPeople
+     * service.
+     */
+    private class ListPeopleResponseSerializer extends AbstractServiceResponseSerializer {
+
+        @Override
+        /**
+         * Serializes the response data.
+         *
+         * @param response ServiceResponse holding the application specific
+         * response object
+         * @param soapResponse SOAPMessage's response object where the response
+         * element is added
+         * @param envelope SOAPMessage's SOAPEnvelope object
+         */
+        public void serializeResponse(ServiceResponse response, SOAPElement soapResponse, SOAPEnvelope envelope) throws SOAPException {
+            // Add "message" element
+            SOAPElement data = soapResponse.addChildElement(envelope.createName("people"));
+            // Get our list of people to serialize into the response and cast it to the "Person" class
+            final List<Person> people = (List<Person>) response.getResponseData();
+            // Loop over every person object and serialize it into our response list
+            for (Person person : people) {
+                serializePerson(envelope, data.addChildElement(envelope.createName("person")), person);
+            }
+        }
+
+        /**
+         * Serializes the {@link Person} object into "shortPerson" as defined in our WSDL
+         * @param envelope SOAPMessage's SOAPEnvelope object
+         * @param parentElement the parent element to add our data under
+         * @param person the {@link Person} object to serialize
+         */
+        private void serializePerson(final SOAPEnvelope envelope, final SOAPElement parentElement, final Person person) throws SOAPException {
+            // Create and add the "SSN" element
+            SOAPElement SSN = parentElement.addChildElement(envelope.createName("ssn"));
```

Comment:
Since `SSN` is a local variable and not a constant, it would be better to rename it to `ssn`.

Reply:
Changed